corva/network/api.py

```diff
@@ -78,35 +83,36 @@ def _get_url(self, path: str):
 
         return os.path.join(base_url.strip('/'), path.strip('/'))
 
-    def _request(
-         self,
-         method: str,
-         path: str,
-         data: Optional[dict] = None,  # request body
-         params: Optional[dict] = None,  # url query string params
-         headers: Optional[dict] = None,  # additional headers to include in request
-         max_retries: Optional[int] = None,  # custom value for max number of retries
-         timeout: Optional[int] = None,  # request timeout in seconds
-    ) -> Response:
-
-        if method not in self.ALLOWED_METHODS:
-            raise ValueError(f'Invalid HTTP method {method}.')
-
-        max_retries = max_retries or self.max_retries
-        timeout = timeout or self.timeout
-
-        # not thread safe
-        self.session.adapters['https://'].max_retries.total = max_retries
-
-        response = self.session.request(
-            method=method,
-            url=self._get_url(path=path),
-            params=params,
-            json=data,
-            headers=headers,
-            timeout=timeout
-        )
+    def _request(self, method: str):
+        def _request_helper(
+             path: str,
+             *,
+             data: Optional[dict] = None,  # request body
+             params: Optional[dict] = None,  # url query string params
+             headers: Optional[dict] = None,  # additional headers to include in request
+             max_retries: Optional[int] = None,  # custom value for max number of retries
+             timeout: Optional[int] = None,  # request timeout in seconds
+        ) -> Response:
+            if method not in self.ALLOWED_METHODS:
```

Comment:
Overcomplicated. We can make do without a nested function

Reply:
I didn't find another way to get proper parameter information like displayed above

Reply:
make user-facing function accept these as parameters?